flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/rank/RetractableTopNFunctionTest.java

```diff
@@ -219,4 +219,38 @@ public void testDisableGenerateRetraction() throws Exception {
 				.assertOutputEqualsSorted("output wrong.", expectedOutput, testHarness.getOutput());
 	}
 
+	@Test
+	public void testCleanIdleState() throws Exception {
+		AbstractTopNFunction func = createFunction(RankType.ROW_NUMBER, new ConstantRankRange(1, 2), true,
+			true);
+		OneInputStreamOperatorTestHarness<BaseRow, BaseRow> testHarness = createTestHarness(func);
+		testHarness.open();
+
+		testHarness.processElement(record("book", 1L, 12));
+		testHarness.processElement(retractRecord("book", 1L, 12));
+		testHarness.processElement(record("fruit", 4L, 33));
+		testHarness.processElement(record("book", 2L, 19));
+		testHarness.processElement(record("fruit", 3L, 22));
+		testHarness.processElement(record("fruit", 5L, 43));
+
+		// cleanup state explicitly
+		func.onTimer(System.currentTimeMillis(), null, null);
```

Comment:
We should not call methods on the function directly. It can't clear the state correctly.
Instead, we should call `testHarness.setProcessingTime` to trigger the `onTimer`.

You can take `OverWindowHarnessTest` as an example.